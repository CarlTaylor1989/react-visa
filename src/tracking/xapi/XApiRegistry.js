/**
 * XApi Registry
 * Activities, extentions and verbs are based on the tincapi registry
 * https://registry.tincanapi.com
 */
const XApiRegistry = {
  activities: {
    assessment: {
      name: 'Assessment',
      id: 'http://adlnet.gov/expapi/activities/assessment'
    },
    'cmi.interactions': {
      name: 'File',
      id: 'http://adlnet.gov/expapi/activities/cmi.interaction'
    },
    course: {
      name: 'Course',
      id: 'http://adlnet.gov/expapi/activities/course'
    },
    game: {
      name: 'game',
      id: 'http://activitystrea.ms/schema/1.0/game'
    },
    goal: {
      name: 'Goal',
      id: 'http://id.tincanapi.com/activitytype/goal'
    },
    interaction: {
      name: 'Interaction',
      id: 'http://adlnet.gov/expapi/activities/interaction'
    },
    link: {
      name: 'Link',
      id: 'http://adlnet.gov/expapi/activities/link'
    },
    media: {
      name: 'Media',
      id: 'http://adlnet.gov/expapi/activities/media'
    },
    meeting: {
      name: 'Meeting',
      id: 'http://adlnet.gov/expapi/activities/meeting'
    },
    module: {
      name: 'Module',
      id: 'http://adlnet.gov/expapi/activities/module'
    },
    objective: {
      name: 'Objective',
      id: 'http://adlnet.gov/expapi/activities/objective'
    },
    performance: {
      name: 'Performance',
      id: 'http://adlnet.gov/expapi/activities/performance'
    },
    question: {
      name: 'Question',
      id: 'http://adlnet.gov/expapi/activities/question'
    },
    simulation: {
      name: 'Simulation',
      id: 'http://adlnet.gov/expapi/activities/simulation'
    },
    slide: {
      name: 'slide',
      id: 'http://id.tincanapi.com/activitytype/slide'
    },
    'slide-deck': {
      name: 'slide-deck',
      id: 'http://id.tincanapi.com/activitytype/slide-deck'
    },
    survey: {
      name: 'survey',
      id: 'http://id.tincanapi.com/activitytype/survey'
    }
  },
  extensions: {
    'browser-info': {
      name: 'browser information',
      id: 'http://id.tincanapi.com/extension/browser-info'
    },
    duration: {
      name: 'duration',
      id: 'http://id.tincanapi.com/extension/duration'
    },
    referrer: {
      name: 'referrer',
      id: 'http://id.tincanapi.com/extension/referrer'
    }
  },
  verbs: {
    achieved: {
      name: 'achieved',
      id: 'https://w3id.org/xapi/dod-isd/verbs/achieved'
    },
    answered: {
      name: 'answered',
      id: 'http://adlnet.gov/expapi/verbs/answered'
    },
    asked: {
      name: 'asked',
      id: 'http://adlnet.gov/expapi/verbs/asked'
    },
    attempted: {
      name: 'attempted',
      id: 'http://adlnet.gov/expapi/verbs/attempted'
    },
    attended: {
      name: 'attended',
      id: 'http://adlnet.gov/expapi/verbs/attended'
    },
    commented: {
      name: 'commented',
      id: 'http://adlnet.gov/expapi/verbs/commented'
    },
    completed: {
      name: 'completed',
      id: 'http://adlnet.gov/expapi/verbs/completed'
    },
    exited: {
      name: 'exited',
      id: 'http://adlnet.gov/expapi/verbs/exited'
    },
    experienced: {
      name: 'experienced',
      id: 'http://adlnet.gov/expapi/verbs/experienced'
    },
    failed: {
      name: 'failed',
      id: 'http://adlnet.gov/expapi/verbs/failed'
    },
    imported: {
      name: 'imported',
      id: 'http://adlnet.gov/expapi/verbs/imported'
    },
    initialized: {
      name: 'initialized',
      id: 'http://adlnet.gov/expapi/verbs/initialized'
    },
    interacted: {
      name: 'interacted',
      id: 'http://adlnet.gov/expapi/verbs/interacted'
    },
    launched: {
      name: 'launched',
      id: 'http://adlnet.gov/expapi/verbs/launched'
    },
    lost: {
      name: 'lost',
      id: 'http://activitystrea.ms/schema/1.0/lose'
    },
    mastered: {
      name: 'mastered',
      id: 'http://adlnet.gov/expapi/verbs/mastered'
    },
    opened: {
      name: 'opened',
      id: 'http://activitystrea.ms/schema/1.0/open'
    },
    passed: {
      name: 'passed',
      id: 'http://adlnet.gov/expapi/verbs/passed'
    },
    preferred: {
      name: 'preferred',
      id: 'http://adlnet.gov/expapi/verbs/preferred'
    },
    progressed: {
      name: 'progressed',
      id: 'http://adlnet.gov/expapi/verbs/progressed'
    },
    rated: {
      name: 'rated',
      id: 'http://id.tincanapi.com/verb/rated'
    },
    received: {
      name: 'received',
      id: 'http://activitystrea.ms/schema/1.0/receive'
    },
    registered: {
      name: 'registered',
      id: 'http://adlnet.gov/expapi/verbs/registered'
    },
    responded: {
      name: 'responded',
      id: 'http://adlnet.gov/expapi/verbs/responded'
    },
    resumed: {
      name: 'resumed',
      id: 'http://adlnet.gov/expapi/verbs/resumed'
    },
    scored: {
      name: 'scored',
      id: 'http://adlnet.gov/expapi/verbs/scored'
    },
    shared: {
      name: 'shared',
      id: 'http://adlnet.gov/expapi/verbs/shared'
    },
    suspended: {
      name: 'suspended',
      id: 'http://adlnet.gov/expapi/verbs/suspended'
    },
    terminated: {
      name: 'terminated',
      id: 'http://adlnet.gov/expapi/verbs/terminated'
    },
    voided: {
      name: 'voided',
      id: 'http://adlnet.gov/expapi/verbs/voided'
    }
  }
};

export default XApiRegistry;
