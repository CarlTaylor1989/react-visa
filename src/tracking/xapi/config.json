{
  "version": "1.0.0",
  "projectId": "mario19-037",
  "gameName": "visa-inside-track",
  "baseURL": "https://lms.visa.com/e-learning/",
  "extensionURL": "https://lms.visa.com/xapi/cornerstone/extensions/",
  "title": "Mario technical prototype",
  "homePage": "https://lms.visa.com",
  "gomoCourseURL": "http://www.gomolearning.com/xapi/activities/course/", 
  "debugMode": false
}
