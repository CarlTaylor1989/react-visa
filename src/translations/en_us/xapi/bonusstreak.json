{
  "title": "Bonus Streak",
  "description": "A bonus streak awarded for successfully completing for the first time four challenges in a row."
}
