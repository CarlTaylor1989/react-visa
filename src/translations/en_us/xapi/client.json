{
  "questionTitle": "%{name} client proposal step %{index}",
  "questionDescription": "%{name} client proposal with topic %{problem}, %{question} selection:",
  "question1": "statement",
  "question2": "product",

  "completedDescription": "A client proposal with topic %{problem}"
}
