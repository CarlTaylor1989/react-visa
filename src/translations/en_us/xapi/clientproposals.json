{
  "title": "Client Proposals page",
  "description": "The page containing the client proposals.",

  "proposalTitle": "Client Proposal %{client}",
  "proposalDescription": "A client proposal in the Visa Inside Track Game",

  "learnMoreTitle": "%{product} Learn more"
}
