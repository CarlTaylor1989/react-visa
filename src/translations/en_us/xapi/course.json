{
  "title": "The Visa Inside Track Game",
  "description": "An experience learners can engage with little and often, as well as for more extended play periods. The game mechanics contain a mixture of short 3-5 minute learning events which can be seen as daily goals, as well as longer term objectives the learner can feel invested in."
}
