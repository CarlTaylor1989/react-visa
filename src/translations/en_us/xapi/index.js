import course from './course.json';
import diagnostic from './diagnostic.json';
import introseq from './introseq.json';
import bonusstreak from './bonusstreak.json';
import ranks from './ranks.json';
import settings from './settings.json';
import map from './map.json';
import products from './products.json';
import dashboard from './dashboard.json';
import gomo from './gomo';
import clientproposals from './clientproposals.json';
import client from './client.json';
import tutorials from './tutorials.json';
import feedback from './feedback.json';

export default {
  course,
  diagnostic,
  introseq,
  bonusstreak,
  ranks,
  settings,
  map,
  products,
  dashboard,
  gomo,
  clientproposals,
  client,
  tutorials,
  feedback
};
