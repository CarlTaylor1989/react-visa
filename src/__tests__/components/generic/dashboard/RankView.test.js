/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import {
  RankView,
  mapStateToProps
} from '../../../../components/generic/dashboard/RankView';
import { sendInteractionData } from '../../../../tracking/xapi/XApiAdapter';

jest.mock('../../../../tracking/xapi/XApiAdapter');

let mockProps;
beforeEach(() => {
  jest.resetAllMocks();
  mockProps = {
    rank: 'r20',
    total: 0
  };
});

test('render RankView component', () => {
  const wrapper = shallow(<RankView {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('method componentDidMount()', () => {
  const wrapper = shallow(<RankView {...mockProps} />);
  const offsetTop = 200;
  const offsetHeight = 67;
  wrapper.instance().rankRef = { current: { offsetTop, offsetHeight } };
  wrapper.instance().componentDidMount();
  expect(wrapper.instance().state.scrollTop).toEqual(offsetTop - offsetHeight);
});

test('method componentWillUnmount()', () => {
  const wrapper = shallow(<RankView {...mockProps} />);
  wrapper.instance().componentWillUnmount();
  expect(sendInteractionData).toHaveBeenCalledWith(expect.objectContaining({
    type: 'experienced',
    activity: expect.objectContaining({
      id: expect.stringContaining('rank'),
      type: 'slide',
      title: expect.any(String),
      description: expect.any(String)
    }),
    response: expect.objectContaining({
      timeAvailable: expect.any(Date),
      timeResponse: expect.any(Date)
    }),
    parent: expect.any(String)
  }));
});

test('should mapStateToProps', () => {
  const state = {
    scoreData: {
      rank: 'r15',
      total: 900000
    }
  };
  const props = mapStateToProps(state);
  expect(props).toEqual({
    ...state.scoreData
  });
});
