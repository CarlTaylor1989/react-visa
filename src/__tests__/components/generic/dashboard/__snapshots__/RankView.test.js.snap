// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render RankView component 1`] = `
<div
  className="rankView"
>
  <Scrollbar
    contentProps={Object {}}
    fallbackScrollbarWidth={20}
    maximalThumbYSize={83}
    minimalThumbSize={30}
    minimalThumbYSize={83}
    momentum={true}
    noDefaultStyles={true}
    scrollDetectionThreshold={100}
    scrollTop={0}
    scrollerProps={Object {}}
    thumbXProps={Object {}}
    thumbYProps={Object {}}
    trackClickBehavior="jump"
    trackXProps={Object {}}
    trackYProps={Object {}}
    wrapperProps={Object {}}
  >
    <div
      className="contentWrapper"
    >
      <div
        className="rankRow "
        key="r1"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r1"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={11000000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r2"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r2"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={10500000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r3"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r3"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={9750000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r4"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r4"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={8500000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r5"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r5"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={7500000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r6"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r6"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={6500000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r7"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r7"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={5550000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r8"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r8"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={4750000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r9"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r9"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={4000000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r10"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r10"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={3500000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r11"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r11"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={2750000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r12"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r12"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={2250000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r13"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r13"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={1750000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r14"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r14"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={1250000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r15"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r15"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={850000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r16"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r16"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={500000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r17"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r17"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={350000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r18"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r18"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={200000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow "
        key="r19"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r19"
          />
          <div
            className="rankRowName"
          >
            <Translate
              tag="span"
              value="generic.dashboard.hidden"
            />
          </div>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={100000}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
      <div
        className="rankRow  current"
        key="r20"
      >
        <div
          className="rankIdentifier"
        >
          <div
            className="rankRowIcon r20"
          />
          <Connect(Tooltip)
            fieldName="generic.tooltips.dashboard.rank.title"
          >
            <div
              className="rankRowName"
            >
              <Translate
                tag="span"
                value="generic.ranks.r20"
              />
            </div>
          </Connect(Tooltip)>
        </div>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.progress"
        >
          <div
            className="progress"
          >
            <ProgressBar
              colour="dark"
              percent={0}
            />
          </div>
        </Connect(Tooltip)>
        <Connect(Tooltip)
          fieldName="generic.tooltips.dashboard.rank.points"
        >
          <div
            className="rankRowPoints"
          >
            <NumberFormat
              allowEmptyFormatting={false}
              allowNegative={true}
              decimalSeparator="."
              displayType="text"
              fixedDecimalScale={false}
              getInputRef={[Function]}
              isAllowed={[Function]}
              isNumericString={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              onValueChange={[Function]}
              prefix=""
              suffix=""
              thousandSeparator=","
              thousandsGroupStyle="thousand"
              type="text"
              value={0}
            />
            <Translate
              tag="span"
              value="generic.dashboard.points"
            />
          </div>
        </Connect(Tooltip)>
      </div>
    </div>
  </Scrollbar>
</div>
`;
